watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -0,0 +1,569 @@
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+import os
+from pyomo.environ import (
+    ConcreteModel,
+    Block,
+    Expression,
+    value,
+    TransformationFactory,
+    units as pyunits,
+    assert_optimal_termination,
+)
+from pyomo.network import Arc, SequentialDecomposition
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+from idaes.core.util.initialization import propagate_state
+
+import idaes.core.util.scaling as iscale
+from idaes.generic_models.unit_models import Mixer, Separator, Product, Feed
+from idaes.generic_models.unit_models.mixer import MomentumMixingType
+from idaes.generic_models.unit_models.translator import Translator
+from idaes.generic_models.costing import UnitModelCostingBlock
+from idaes.core.util.exceptions import ConfigurationError
+
+from watertap.unit_models.pressure_exchanger import PressureExchanger
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
+from watertap.core.util.initialization import assert_degrees_of_freedom
+
+import watertap.property_models.seawater_prop_pack as prop_SW
+from watertap.unit_models.reverse_osmosis_0D import (
+    ReverseOsmosis0D,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.core.wt_database import Database
+import watertap.core.zero_order_properties as prop_ZO
+from watertap.unit_models.zero_order import (
+    FeedZO,
+    PumpElectricityZO,
+    NanofiltrationZO,
+    SecondaryTreatmentWWTPZO,
+)
+from watertap.core.zero_order_costing import ZeroOrderCosting
+from watertap.costing import WaterTAPCosting
+
+
+def main():
+    m = build()
+    set_operating_conditions(m)
+
+    assert_degrees_of_freedom(m, 0)
+    assert_units_consistent(m)
+
+    initialize_system(m)
+    assert_degrees_of_freedom(m, 0)
+
+    results = solve(m)
+
+    add_costing(m)
+    initialize_costing(m)
+    assert_degrees_of_freedom(m, 0)
+
+    solve(m)
+    # display_results(m)
+    display_costing(m)
+
+    return m, results
+
+
+def build():
+    # flowsheet set up
+    m = ConcreteModel()
+    m.db = Database()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    # define property packages
+    m.fs.prop_nf = prop_ZO.WaterParameterBlock(default={"solute_list": ["dye", "tds"]})
```

Comment:
Technically, since we are adding the secondary WWTP step, we could add `bod `and `tss` to the feed, and these would be removed by the WWTP. However, at this level, the only significance the addition of the WWTP step has is accounting for the extra treatment costs, so I suppose we could omit the addition of bod and tss in the feed. It might complicate things with regard to the translator block.